### PR TITLE
functional syr2k and her2k

### PR DESCRIPTION
**Pull request by TorreZuk**
Summary of proposed changes:
- functional syr2k and her2k
- removed /efi partition formatting feature
- Daine I think your commit was temporarily added in a rebase that is why you are in the commit

### CODE REVIEW

File: clients/gtest/her2k_gtest.cpp [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+{
+    // possible gemv test cases
+    enum her2k_test_type
+    {
+        HER2K,
+        HER2K_BATCHED,
+        HER2K_STRIDED_BATCHED,
+    };
+
+    //ger test template
```
Comment (daineAMD) [inline clients/gtest/her2k_gtest.cpp]: Update this comment and the previous gemv one :)

File: clients/gtest/syr2k_gtest.cpp [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+#include "testing_syr2k.hpp"
+#include "testing_syr2k_batched.hpp"
+#include "testing_syr2k_strided_batched.hpp"
+#include "type_dispatch.hpp"
+#include <cstring>
+#include <type_traits>
+
+namespace
+{
+    // possible gemv test cases
```
Comment (daineAMD) [inline clients/gtest/syr2k_gtest.cpp]: Same here

File: clients/gtest/her2k_gtest.cpp [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+            name << rocblas_datatype2string(arg.a_type);
+
+            if(strstr(arg.function, "_bad_arg") != nullptr)
+            {
+                name << "_bad_arg";
+            }
+            else
+            {
+                // alpha & beta only real
```
Comment (daineAMD) [inline clients/gtest/her2k_gtest.cpp]: Only beta is real in her2k right? Lines 87-89 can be removed, but it's insignificant.

File: library/src/blas3/rocblas_syr2k.hpp [hunk 4]
```diff
@@ -0,0 +1,450 @@
...
+                                       TPtr              CP,
+                                       rocblas_int       offsetC,
+                                       rocblas_int       ldc,
+                                       rocblas_stride    strideC,
+                                       rocblas_int       batch_count)
+{
+    if(uplo != rocblas_fill_lower && uplo != rocblas_fill_upper)
+        return rocblas_status_invalid_value;
+
+    if(trans != rocblas_operation_none && trans != rocblas_operation_transpose)
```
Comment (daineAMD) [inline library/src/blas3/rocblas_syr2k.hpp]: See comment in rocblas-functions.h

File: library/include/rocblas-functions.h [hunk 5]
```diff
@@ -9498,6 +9834,432 @@ ROCBLAS_EXPORT rocblas_status rocblas_zsyrk_strided_batched(rocblas_handle
...
+    handle    [rocblas_handle]
+              handle to the rocblas library context queue.
+
+    @param[in]
+    uplo    [rocblas_fill]
+            rocblas_fill_upper:  C is an upper triangular matrix
+            rocblas_fill_lower:  C is a  lower triangular matrix
+
+    @param[in]
+    trans  [rocblas_operation]
```
Comment (daineAMD) [inline library/include/rocblas-functions.h]: Oddly enough, in the ssyr2k documentation on netlib, they allow a conjugate-transpose operation, presumably this does the same as the simple transpose operation. For the complex cases (csyr2k, eg.) they do not support conjugate-transpose. Seems a little odd, not sure if we should support that as well.

File: library/src/blas3/rocblas_syr2k.hpp [hunk 6]
```diff
@@ -0,0 +1,450 @@
...
+{
+    __shared__ T atile[TILE_NK][TILE_NK];
+    __shared__ T btile[TILE_NK][TILE_NK];
+
+    int col_pos = blockIdx.y * TILE_NK;
+    int row_pos = blockIdx.x * TILE_NK;
+
+    int tilefrom = upper ? row_pos : col_pos;
+    int tileto   = upper ? col_pos : row_pos;
+    if(!alpha || tilefrom > tileto)
```
Comment (daineAMD) [inline library/src/blas3/rocblas_syr2k.hpp]: Shouldn't need !alpha check here.

File: library/include/rocblas-functions.h (hunk 5 above)
Comment (TorreZuk) [reply library/include/rocblas-functions.h]: Yes that is in the documentation but the code only does simple transpose and non-transpose.  As the other implementations don't mention it (e.g magmaTrans & magmaNoTrans) I think it is better to keep it clear this doesn't do conjugate transpose as Hermitian function her2k handles this.

**Pull request merged**